### DISCUSSION TITLE
Modify Bleachbit Cleaners with Windows [Bad IDEA maybe ?]

### DISCUSSION THREAD

**Discussion by rotckej** [bleachbit/bleachbit#1430] [Q&A]
Hi Everyone ! 🦊 

I search since 2 days before coming here (hope this question is not already answered ...) 😅 

I download the source code to modify some **xml** cleaners and then "**rebuild**" the bleachbit.py to get a modified one. 

I also tried on a **Ubuntu 22.04** (and it worked !! 👍 )

But ... as **always** on Windows this doesn't work so easily 👯 

I tried some things : 
- Download package on PyCharm 
- Get the MINGW64 to open .py and stuff 
- install all module defined on the BleachBit source code
- deleting some code to pass the error ^^'...
- Tried on VM too (to get a fresh install)

Some Screens 📷 : 
1. Launch from PyCharm (same result with Powershell) :
![image](https://user-images.githubusercontent.com/26737678/203272982-d4c2813a-3dcb-477d-9a5e-d3f7946ee257.png)
PS : I follow the page to install PyGObject ^^...

2. Launch from MINGW64 window :
![image](https://user-images.githubusercontent.com/26737678/203273098-fbe3a765-0398-4161-a061-f74a5c37222f.png)

3. Windows Setup ?  Its a no...
![image](https://user-images.githubusercontent.com/26737678/203277545-ceef6fae-71ec-43fe-9b5c-65707cfb6f7f.png)

4. Make -C po local 
![image](https://user-images.githubusercontent.com/26737678/203273852-5c16e721-f007-4d0c-be81-0f493ed5f944.png)

Thanks in advance for help 🦊 

Qwyn

**Comment by az0** [comment]
> I download the source code to modify some **xml** cleaners and then "**rebuild**" the bleachbit.py to get a modified one.

You can edit the CleanerML (XML) files without rebuilding, recompiling, or repacking. Just drop them in one of the [relevant directories](https://docs.bleachbit.org/cml/cleanerml.html#storage), and restart the application.

In case of the Windows portable version, you can repackage just by recreating a zip file. 

> But ... as always on Windows this doesn't work so easily dancers

To rebuild the Windows package, follow  the appveyor.xml file. Today there are quite different versions across branches: master vs python10-mkhon branch.

**Reply by rotckej** [reply]
Thanks for the answer --> I'll test it ASAP :)
